### PR TITLE
Create custom intl twig functions

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

There are conflicts using the last version of this bundle and `SonataIntlBundle` because both register twig functions with the same name and different signature. 

The only way I could come up with is removing `twig/extra-bundle`, so the `IntlExtension` from `twig/intl-extra` is not automatically registered anymore and I created our `IntlExtension` which decorates the Twig one and exposes only the methods we're using changing their names to `sonata_*`. 

The problem I see now is people using these functions in their projects and we should remove the in the next major version since this is a patch to make it work with `sonata/intl-bundle`. Don't know if there is a way to make twig functions as internal or something like that.

I am targeting this branch, because this is BC.

Closes https://github.com/sonata-project/SonataIntlBundle/issues/299

## Changelog

```markdown
### Removed
- `twig/extra-bundle` dependency
### Fixed
- Compatibility with `sonata/intl-bundle`
```

To test:

```shell
composer config repositories.franmomu vcs https://github.com/franmomu/SonataAdminBundle
composer require sonata-project/admin-bundle "dev-allow_intl_extension as 3.62.1"
```

### CODE REVIEW

File: src/DependencyInjection/SonataAdminExtension.php [hunk 1]
```diff
@@ -94,6 +97,14 @@ public function load(array $configs, ContainerBuilder $container)
             if ('@SonataAdmin/CRUD/history_revision_timestamp.html.twig' === $config['templates']['history_revision_timestamp']) {
                 $config['templates']['history_revision_timestamp'] = '@SonataAdmin/CRUD/Intl/history_revision_timestamp.html.twig';
             }
+
+            $twigIntlExtension = new Definition(TwigIntlExtension::class);
```
Comment (core23) [inline src/DependencyInjection/SonataAdminExtension.php]: Can you please try to extract some small methods. This will improve readability.

**Comment by phansys** [comment]
Great job @franmomu!
IMO, there is a missing part yet. The Twig functions at "sonata-project/intl-bundle" are using helpers to infer the locale and the timezone, in order to provide these values as arguments for the Intl core methods.
If those values are not passed anymore, the results will be different where the values were different than the default.
In that case, the users will be forced to update their codebase in order to provide these values manually each time the functions are called.

**Comment by franmomu** [comment]
Taking a look at this, we would need to import all the [Timezone](https://github.com/sonata-project/SonataIntlBundle/tree/2.x/src/Timezone) and [Locale](https://github.com/sonata-project/SonataIntlBundle/tree/2.x/src/Locale) classes. Add [the `timezone` configuration](https://github.com/sonata-project/SonataIntlBundle/blob/2.x/src/DependencyInjection/Configuration.php#L46) to AdminBundle and in case in the `SonataIntlBundle` is enabled and there is no configuration in `SonataAdminBundle`, it should use the `SonataIntlBundle` one to be BC. And also decide what to do with [UserBasedTimezoneDetector](https://github.com/sonata-project/SonataIntlBundle/blob/2.x/src/Timezone/UserBasedTimezoneDetector.php).

So... at this point I don't know if it's better to just revert https://github.com/sonata-project/SonataAdminBundle/pull/5835 and make all these changes in `master` or keep patching things.

**Comment by phansys** [comment]
>So... at this point I don't know if it's better to just revert #5835 and make all these changes in master or keep patching things.

The revert of #5835 has my vote :+1:

**Pull request closed without merging**